cmd/web.go

```diff
@@ -81,6 +86,22 @@ func runHTTPRedirector() {
 	}
 }
 
+func createPIDFile(pidPath string) {
+	currentPid := os.Getpid()
+	if err := os.MkdirAll(filepath.Dir(pidPath), os.ModePerm); err != nil {
+		log.Fatal("Failed to create PID folder: %v", err)
+	}
+
+	file, err := os.Create(pidPath)
+	if err != nil {
+		log.Fatal("Failed to create PID file: %v", err)
+	}
+	defer file.Close()
+	if _, err := file.WriteString(strconv.FormatInt(int64(currentPid), 10)); err != nil {
+		log.Fatal("Failed to write PID information: %v", err)
+	}
```

Comment:
I know it's copied from the old code, but I think this is a chance to simplify it:
```suggestion
	if err := os.WriteFile(pidPath, fmt.Appendf(nil, "%d", pid), 0o666); err != nil {
		log.Fatal("Failed to write PID information: %v", err)
	}
```
Just a suggestion, no blocking.

Reply:
The logic has a slight difference. In the original logic, creating the pid file failure will result in a fatal exit but `os.WriteFile` will not. So I will keep this as before since it's a refactor.

Reply:
>  creating the pid file failure will result in a fatal exit but `os.WriteFile` will not.

I don't think so, `os.WriteFile` will return an error if cannot create the file, and will result in a fatal exit too.

> keep this as before since it's a refactor.

Agree.